src/ui/Base/QueryBuilder.ts

```diff
@@ -499,4 +506,20 @@ export class QueryBuilder {
     const query = this.build();
     return Utils.isNonEmptyString(query.q) || Utils.isNonEmptyString(query.lq);
   }
+
+  private validateGroupBy() {
+    if (!this.groupByRequests.length && this.facets.length) {
+      return undefined;
```

Comment:
Should we return `undefined` or an empty array here? Same for line 519.

Reply:
They have to be undefined, sending both will result in an API error

Reply:
Supporting both queries on the API side will require us to support both types of facets in the UI at the same time, which I cannot guarantee will work flawlessly OOTB. It's safer to prevent both params on our side until we're sure to be able to support that feature.

Side note: the error msg is clear from the API when we send both, it will be easy for a client using a custom UI there's no issue there.